geode-core/src/main/java/org/apache/geode/admin/internal/InetAddressUtil.java

```diff
@@ -1,200 +0,0 @@
-    } catch (java.net.UnknownHostException e) {
-      logStackTrace(e);
-      Assert.assertTrue(false, "Failed to get local host");
-      return null; // will never happen
-    }
-  }
-
-  /**
-   * Validates the host by making sure it can successfully be used to get an instance of
-   * InetAddress. If the host string is null, empty or would result in
-   * <code>java.lang.UnknownHostException</code> then null is returned.
-   * <p>
-   * Any leading slashes on host will be ignored.
-   *
-   * @param host string version the InetAddress
-   * @return the host converted to InetAddress instance
-   */
-  public static String validateHost(String host) {
-    if (host == null || host.length() == 0) {
-      return null;
-    }
-    try {
-      InetAddress.getByName(trimLeadingSlash(host));
-      return host;
-    } catch (java.net.UnknownHostException e) {
-      logStackTrace(e);
-      return null;
-    }
-  }
-
-  /** Returns true if host matches the LOCALHOST. */
-  public static boolean isLocalHost(Object host) {
-    if (host instanceof InetAddress) {
-      if (LOCALHOST.equals(host)) {
-        return true;
-      } else {
-        try {
-          Enumeration en = NetworkInterface.getNetworkInterfaces();
-          while (en.hasMoreElements()) {
-            NetworkInterface i = (NetworkInterface) en.nextElement();
-            for (Enumeration en2 = i.getInetAddresses(); en2.hasMoreElements();) {
-              InetAddress addr = (InetAddress) en2.nextElement();
-              if (host.equals(addr)) {
-                return true;
-              }
-            }
-          }
-          return false;
-        } catch (SocketException e) {
-          throw new GemFireIOException(
-              "Unable to query network interface",
-              e);
-        }
-      }
-    } else {
-      return isLocalHost(InetAddressUtil.toInetAddress(host.toString()));
-    }
-  }
-
-  /** Returns true if host matches the LOOPBACK (127.0.0.1). */
-  public static boolean isLoopback(Object host) {
-    if (host instanceof InetAddress) {
-      return LOOPBACK.equals(host);
-    } else {
-      return isLoopback(InetAddressUtil.toInetAddress(host.toString()));
-    }
-  }
-
-  /** Returns a version of the value after removing any leading slashes */
-  private static String trimLeadingSlash(String value) {
-    if (value == null)
-      return "";
-    while (value.indexOf("/") > -1) {
-      value = value.substring(value.indexOf("/") + 1);
-    }
-    return value;
-  }
-
-  /**
-   * Logs the stack trace for the given Throwable if logger is initialized else prints the stack
-   * trace using System.out. If logged the logs are logged at WARNING level.
-   *
-   * @param throwable Throwable to log stack trace for
-   */
-  private static void logStackTrace(Throwable throwable) {
-    AdminDistributedSystemImpl adminDS = AdminDistributedSystemImpl.getConnectedInstance();
-
-    logger.warn(throwable.getMessage(), throwable);
-  }
```

Comment:
It looks like the old version of this class used to log warnings under certain error conditions but the new version of the class does not. Are we sure nobody is relying on these warnings being logged?

Reply:
No one is using Admin API. It was deprecated before Apache Geode 1.0.0. I removed the logging of stack traces mainly because it's bad form for simple low-level utils like this to perform logging. The main reason to move this class out of admin.internal is because we're going to delete everything under admin.* but we have lots of code outside of admin.* using this class. I could add the logging back in if you feel strongly about it, but I don't think it's valuable.

If we change it further, then my preference would be to propagate the caught exception instead of logging it:
```
} catch (UnknownHostException e) {
  throw new AssertionError("Failed to get InetAddress: " + host**, e**);
}
```

Reply:
My concern was that we might be removing logging that users rely on. If we're sure that this logging isn't valuable/useful then I'm fine with removing it.

Reply:
I added InetAddressUtilsWithLogging which wraps two of the utils with versions that log call stack. I also updated all uses of both InetAddressUtils classes to use `import static`